pkg/cache/cluster.go

```diff
@@ -846,14 +846,15 @@ func (c *clusterCache) IterateHierarchy(key kube.ResourceKey, action func(resour
 					return strings.Compare(key1.String(), key2.String()) < 0
 				})
 				child := children[0]
-				action(child, nsNodes)
-				child.iterateChildren(nsNodes, map[kube.ResourceKey]bool{res.ResourceKey(): true}, func(err error, child *Resource, namespaceResources map[kube.ResourceKey]*Resource) {
-					if err != nil {
-						c.log.V(2).Info(err.Error())
-						return
-					}
-					action(child, namespaceResources)
-				})
+				if action(child, nsNodes) {
```

Comment:
`action()` is called on line 831, but I don't see the return value being checked. Is that expected?

Reply:
Thank you! This is a bug. Added missing check and unit test